sycl/test-e2e/InvokeSimd/Feature/invoke_simd_struct.cpp

```diff
@@ -53,58 +42,76 @@ using namespace sycl::ext::oneapi::experimental;
-__attribute__((always_inline)) esimd::simd<int, VL>
-ESIMD_CALLEE(int *A, esimd::simd<int, VL> b, int i,
-             multipliers *scalars) SYCL_ESIMD_FUNCTION {
-  esimd::simd<int, VL> a;
+template <class TZ, class TY, class TX> struct trio {
+  TZ z;
+  TY y;
+  TX x;
+};
+
+template <class StructTY>
+__attribute__((always_inline)) esimd::simd<float, VL>
+ESIMD_CALLEE(float *A, int i, StructTY S) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> a;
   a.copy_from(A + i);
-  // Add vectors A + B and scale them by x and y.
-  return a + b * scalars->x * scalars->y;
+  return a * S.x;
 }
 
+template <class StructTY>
 [[intel::device_indirectly_callable]] SYCL_EXTERNAL
-    simd<int, VL> __regcall SIMD_CALLEE(int *A, simd<int, VL> b, int i,
-                                        multipliers *scalars)
-        SYCL_ESIMD_FUNCTION;
-
-int SPMD_CALLEE(int *A, int b, int i, multipliers *scalars) {
-  return A[i] + b * scalars->x * scalars->y;
+    simd<float, VL> __regcall SIMD_CALLEE(float *A, int i,
+                                          StructTY S) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> res = ESIMD_CALLEE<StructTY>(A, i, S);
+  return res;
 }
 
-using namespace sycl;
+enum StructsTypes { Solo = 1, Duo = 2, Trio = 3, DuoSameType = 4 };
 
-constexpr bool use_invoke_simd = true;
+template <int> class TestID;
 
-int main(void) {
+using namespace sycl;
+
+template <StructsTypes UsedStruct, class Queue> bool test(Queue q) {
   constexpr unsigned Size = 1024;
   constexpr unsigned GroupSize = 4 * VL;
 
-  auto q = queue{gpu_selector_v};
   auto dev = q.get_device();
-  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
-            << "\n";
   auto ctx = q.get_context();
 
-  int *A = static_cast<int *>(malloc_shared(Size * sizeof(int), dev, ctx));
-  int *B = static_cast<int *>(malloc_shared(Size * sizeof(int), dev, ctx));
-  int *C = static_cast<int *>(malloc_shared(Size * sizeof(int), dev, ctx));
+  float *A =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctx));
+  float *C =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctx));
 
   for (unsigned i = 0; i < Size; ++i) {
-    A[i] = B[i] = i;
+    A[i] = i;
     C[i] = -1;
   }
 
-  // USM shared memory allocation for a struct multipliers.
-  multipliers *scalars = static_cast<multipliers *>(
-      malloc_shared(/*Size **/ sizeof(multipliers), dev, ctx));
-  scalars->x = 2;
-  scalars->y = 3;
+  // solo case, declared outside parallel_for
+  // could be represented internally as a basic type value
+  auto uno = solo<float>{-1.0};
+
+  // duo same type case, declared inside parallel_for
+  // could be represented internally as an array
+  float X = -1.0;
+  float Y = 2;
+
+  // duo case, declared outside parallel_for
+  // could be represented internally as a tuple
+  auto dos = duo<char, float>{2, -1.0};
+
+  // trio case, declared outside parallel_for
+  // not quite sure how exactly this one will be represented internally
```

Comment:
Unrelated to this PR, but I don't see why tuple would work for line 110 yet not work here.